src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights/ApplicationInsightsTelemetryPublisher.cs

```diff
@@ -42,6 +42,11 @@ public ValueTask PublishEvent(EvaluationEvent evaluationEvent, CancellationToken
                 { "Enabled", evaluationEvent.Enabled.ToString() }
             };
 
+            if (evaluationEvent.TargetingContext != null)
+            {
+                properties["TargetingId"] = evaluationEvent.TargetingContext.UserId;
```

Comment:
I am ok with the current implementation. Just curious about what will happen if the TargetingContext.UserId is null.
How will the experimentation link be established in this case? Will all metrics and events with a null "TargetingId" be linked together?

Reply:
That's a good question. The metrics will still be emitted, but linking them together from a null targetingId would not provide much value. Split will ignore the data for Experimentation if the id is null. The only value of the field is to explicitly indicate that there was no value, so it cannot be associated with metrics.